### PR TITLE
feat: automate locations - modified API responses

### PR DESCRIPTION
**Pull request by cka-y**
**Summary:**

This PR updates the feed endpoints and the search endpoint to use the extracted locations and includes a new `country` field for each location within the `gtfs_feeds` response.

**Out of Scope:** 
- Filtering behaviour should use the english version of the locations in #662 (this is also needed to update the integration tests)

**Expected Behavior:**
- Each location in the response now includes a `country` field.
- All location fields are provided in English.

**Testing Tips:**
1. Access the UI link provided in the PR comments.
2. Navigate through the feeds.
3. Inspect the network tab response to verify the presence of the `country` field.
4. Ensure that all location fields are in English.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: integration-tests/src/endpoints/gtfs_feeds.py [hunk 1]
```diff
@@ -26,21 +26,21 @@ def test_gtfs_feeds(self):
...
-    #     )
-    #     for i, country_code in enumerate(country_codes):
-    #         self._test_filter_by_country_code(
-    #             country_code,
-    #             "v1/gtfs_feeds",
-    #             validate_location=True,
-    #             task_id=task_id,
-    #             index=f"{i + 1}/{len(country_codes)}",
-    #         )
+    def test_filter_by_country_code_gtfs(self):
```
Comment (cka-y) [inline integration-tests/src/endpoints/gtfs_feeds.py:29]: This test should still work as the country codes are the same

**Comment by cka-y** [comment]
The integration tests are expected to fail as i have introduced a new view that is not in QA yet

File: liquibase/changes/feat_622.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
+DROP VIEW IF EXISTS location_with_translations;
+CREATE VIEW location_with_translations AS
```
Comment (davidgamez) [inline liquibase/changes/feat_622.sql:2]: If we are creating a view for translations limited to _en_, let's reflect this on the name, something like `location_with_translations_en`.

File: docs/DatabaseCatalogAPI.yaml [hunk 3]
```diff
@@ -677,6 +677,10 @@ components:
             For a list of valid codes [see here](https://unece.org/trade/uncefact/unlocode-country-subdivisions-iso-3166-2).
           type: string
           example: US
+        country:
+            description: The name of the country where the system is located.
```
Comment (davidgamez) [inline docs/DatabaseCatalogAPI.yaml:681]: [suggestion]: Add a note to clarify that this field only supports English translations for now.

**Comment by davidgamez** [comment]
Let's ignore/comment on the integration test to avoid data updates in PROD.

File: docs/DatabaseCatalogAPI.yaml (hunk 3 above)
Comment (cka-y) [reply docs/DatabaseCatalogAPI.yaml:681]: this would apply to each field in the location -- should i specify it for each?

Comment (davidgamez) [reply docs/DatabaseCatalogAPI.yaml:681]: You are right; except country_code, I think is a good idea to be explicit about it as the data in the catalog repo might be in a different language(with the exception of the feeds without datasets that they will have the same value)

**Comment by cka-y** [comment]
> Let's ignore/comment on the integration test to avoid data updates in PROD.

The integration tests that might cause issues post merge are already commented

File: api/src/feeds/impl/feeds_api_impl.py [hunk 4]
```diff
@@ -164,7 +183,7 @@ def get_gtfs_feeds(
         dataset_latitudes: str,
         dataset_longitudes: str,
         bounding_filter_method: str,
-    ) -> List[GtfsFeed]:
+    ) -> List[Gtfsfeed]:
```
Comment (davidgamez) [inline api/src/feeds/impl/feeds_api_impl.py:186]: This should be the API model, not the DB one.

**Review by davidgamez — APPROVED** [review]
👏

**Pull request merged**